File: src/extractorpacman/LogFile.java

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package extractorpacman;

import static engine.pacman.game.Constants.pathMazes;
import engine.pacman.game.internal.Node;
import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.IOException;
import java.io.LineNumberReader;
import java.util.Scanner;

/**
 *
 * @author giang-rocker
 */
public class LogFile {

    String fileName;
    int index;
    int currentTimeStep =0;
    int numOfStage = 0;
    final String pathLogGame = "/engine/data/logGame";
    String gameStage[];
    public LogFile () {
        currentTimeStep = 0;
    }
    
    public String[] getLogFile(String _fileName) throws FileNotFoundException, IOException {
        this.fileName = _fileName;
        gameStage = new String[24000];
        System.out.println((pathLogGame + "/" + fileName + ".txt").toString());
        Scanner scanner = new Scanner(getClass().getResourceAsStream(pathLogGame + "/" + fileName + ".txt"));
         String input ;

     
        
        int index =0 ;
        while (scanner.hasNextLine()) {
            input = scanner.nextLine();
            gameStage[index++] = input;
            
        }
        
        numOfStage = index;
        
        scanner.close();

        return gameStage;
    }
    
    public String getNextStage () {
        if (currentTimeStep < numOfStage) return gameStage[currentTimeStep++];
        else return null;
    }
    
     public String getCurrentState () {
        if (currentTimeStep < numOfStage) return gameStage[currentTimeStep];
        else return null;
    }
}

File: src/extractorpacman/extractorPython/LogFile.py
#!/usr/bin/env python2
#encoding: UTF-8

# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.


class LogFile:
    
    def __init__(self,_logFileName):
        self.logFileName= _logFileName
        self.gameState = self.get_all_game_state()
        self.currentTimeStep = 0
        self.totalGameState = len (self.gameState)
        
    def get_all_game_state(self):
        logFile = open(self.logFileName+'.txt', 'r')
        listGameState=logFile.readlines()
        listGameState.pop(len(listGameState)-1)
        return listGameState
    
    def is_end(self):
        return (self.currentTimeStep == len(self.gameState))
    
    def is_valid(self):
        return (self.currentTimeStep < len(self.gameState)) and (self.currentTimeStep>=0) 
        
    def get_next_state(self):
        self.currentTimeStep+=1
        return self.gameState[self.currentTimeStep-1]
    
    def get_prev_state(self):
        self.currentTimeStep-=1
        return self.gameState[self.currentTimeStep+1]
        
    def get_state(self,_id):
        return self.gameState[_id]
